### PR TITLE
adding the required files for static page verification of CFT

### PR DESCRIPTION
**Pull request by sshajiya**

### CODE REVIEW

File: .github/workflows/static_page_verification.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{secrets.AWS_ACCESS_KEY_ID}}
+          aws-secret-access-key: ${{secrets.AWS_SECRET_ACCESS_KEY}}
+          aws-session-token: ${{secrets.AWS_SESSION_TOKEN}}
+          aws-region: ${{secrets.AWS_REGION}}
+      - name: Create Stack
+        uses: aws-actions/aws-cloudformation-github-deploy@v1
+        with:
+          name: ${{secrets.STACK_NAME}}
```
Comment (464d41) [inline .github/workflows/static_page_verification.yml:28]: Stack name is not a secret. Why you don't hardcode some prefix?

File: .github/workflows/static_page_verification.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+name: Cloud Formation Template Static Page verification
```
Comment (464d41) [inline .github/workflows/static_page_verification.yml:1]: Change destination branch from master to dev. Make sure your code works with latest template from dev branch.

File: .github/workflows/static_page_verification.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        uses: aws-actions/aws-cloudformation-github-deploy@v1
+        with:
+          name: ${{secrets.STACK_NAME}}
+          template: ${{secrets.Template_Name}}
+          capabilities: CAPABILITY_NAMED_IAM
+      - name: Customize the stack for static page verification
+        run: |
+          git clone codecommit://CFT-NAP-TEST-AppProtectRepo local-dir
+          cd local-dir
+          pwd
```
Comment (464d41) [inline .github/workflows/static_page_verification.yml:35]: Debug messages should not to to the final workflow.

File: .github/workflows/static_page_verification.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          cp  $PWD/scripts/appspec.yml $PWD/
+          git config --global user.email ${{secrets.USER_NAME}}
```
Comment (464d41) [inline .github/workflows/static_page_verification.yml:37]: Can username be just a random string? If so just hardcode.

File: scripts/pipeline/appspec.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+version: 0.0
```
Comment (464d41) [inline scripts/pipeline/appspec.yml:1]: Can you place all workflow related files to ".github/workflows" folder?

File: scripts/pipeline/delete_stack.py [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import boto3
```
Comment (464d41) [inline scripts/pipeline/delete_stack.py:1]: Please use "aws-actions" to delete the stack.

File: scripts/pipeline/index.html [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<html>
```
Comment (464d41) [inline scripts/pipeline/index.html:1]: Similar to appspec.yaml. Would be good if all workflow related files move to ".github/workflow" folder.

File: scripts/pipeline/utils.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import boto3
```
Comment (464d41) [inline scripts/pipeline/utils.py:1]: Would be good if all workflow related files move to ".github/workflow" folder.

File: scripts/pipeline/verify_cft_nap.py.py [hunk 9]
```diff
@@ -0,0 +1,25 @@
+import boto3
```
Comment (464d41) [inline scripts/pipeline/verify_cft_nap.py.py:1]: Would be good if all workflow related files move to ".github/workflow" folder.

File: scripts/pipeline/verify_cft_nap.py.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+import os
```
Comment (464d41) [inline scripts/pipeline/verify_cft_nap.py.py:2]: Remove extra ".py" from file extension.

**Review by 464d41 — CHANGES_REQUESTED** [review]

**Comment by 464d41** [comment]
Why did u removed appspec.yaml and index.html files from the PR? They still seem to be required.

**Comment by 464d41** [comment]
Please give a link to this workflow results with latest template in dev branch.

**Comment by sshajiya** [comment]
> 

https://github.com/sshajiya/aws-waf-solution-template/runs/4308204639?check_suite_focus=true

**Comment by sshajiya** [comment]
> Please give a link to this workflow results with latest template in dev branch.
https://github.com/sshajiya/aws-waf-solution-template/runs/4308204639?check_suite_focus=true
The above link is the passed workflow log with the existed template in master branch.

With the new template in dev branch is it failing with default parameter values and asking the new sshkey for the ec2 instance.
Here is the link of log result for reference.

https://github.com/sshajiya/aws-waf-solution-template/runs/4308853375?check_suite_focus=true

**Comment by 464d41** [comment]
> With the new template in dev branch is it failing with default parameter values and asking the new sshkey for the ec2 instance.

In order to merge your test into master it has to run successfully with latest template version from dev branch. Please make it work and update PR accordingly.

**Pull request closed without merging**